### DISCUSSION TITLE
Are RC5 and MDC2 algorithms are disabled by default in OpenSSL 3.3.0?

### DISCUSSION THREAD

**Discussion by caglarkarahan** [openssl/openssl#24720] [Q&A] [answered]
Previously, RC5 and MDC2 algorithms are disabled by default unless "enable-rc5" and "enable-mdc2" are specified.

In OpenSSL version 3.3.0, legacy ciphers are enabled with functions as specified in guideline. With enabling legacy provider when building the openssl or defined for other OSS functionalities, are these algorithms still disabled by default?

**Comment by mnMomma** [comment]
#24607 

- [ ] 
<details><summary>Details</summary>
<p>



</p>
</details>

**Comment by levitte** [comment, marked as answer]
RC5 is still disabled by default, i.e. will not even be available in the legacy provider unless you configure with `enable-rc5`.

MDC2, is enabled by default...  still via the legacy provider, though.  However, there's a catch!  If DES is disabled, MDC2 gets disabled too.

**Comment by mattcaswell** [comment]
RC5 continues to be disabled by default even if using the legacy provider. To use it you must explicitly enable rc5 at openssl compile time. If enabled at compile time low level (deprecated) RC5 APIS are available directly in libcrypto. If you want to use the EVP APIs then you must first additionally load the legacy provider.

MDC2 is not disabled by default (and I don't think it ever has been).

**Comment by caglarkarahan** [comment]
Thanks for quick responses. What about "no-<algo>" option in Configure command when building openssl? 
For example; 

./Configure no-mdc2 no-rc5 enable-weak-ssl-ciphers

Command works fine and take the args as configuration. Is it possible to ignore mdc2 and rc5 and also enable legacy provider with this configuration?

**Reply by paulidale** [reply]
Yes.  Have a look at the migration guide for details.
